python/feldera/pipeline.py

```diff
@@ -118,6 +118,7 @@ def input_json(self, table_name: str, data: Dict | list, update_format: str = "r
     def listen(self, view_name: str) -> OutputHandler:
         """
         Listen to the output of the provided view so that it is available in the notebook / python code.
+        When the pipeline is shutdown, these listeners are dropped.
```

Comment:
what happens if the pipeline is shut down by someone else, e.g., the UI?

Reply:
If the pipeline is shutdown from the UI, the listener stops and returns. 

From the python program's point of view, calling `.start()` again after having called it before is weird and will likely still pause. 

However, if `.restart()` is called, it attempts to shutdown the pipeline first, meaning the listeners are dropped again, and should work as intended.